src/dbt_client/dbtIntegration.ts

```diff
@@ -264,12 +301,16 @@ export interface DBTProjectIntegration extends Disposable {
   ): Promise<{ [key: string]: string }[]>; // TODO: this should be typed
   getColumnsOfModel(modelName: string): Promise<{ [key: string]: string }[]>; // TODO: this should be typed
   getCatalog(): Promise<{ [key: string]: string }[]>; // TODO: this should be typed
+  getDebounceForRebuildManifest(): number;
 }
 
 @provide(DBTCommandExecutionInfrastructure)
 export class DBTCommandExecutionInfrastructure {
-  private queue: DBTCommandExecution[] = [];
-  private running = false;
+  private queues: Map<string, DBTCommandExecution[]> = new Map<
+    string,
+    DBTCommandExecution[]
+  >();
+  private queueStates: Map<string, boolean> = new Map<string, boolean>();
```

Comment:
it will be useful if the value for each key is json object with field like status? If we want to extend this map for handling other values related to execution, it will be easy to add more fields.